src/routeGeneration/templateHelpers.ts

```diff
@@ -359,6 +363,95 @@ export class ValidationService {
     return new Buffer(value);
   }
 
+  public validateUnion(name: string, value: any, fieldErrors: FieldErrors, subSchemas: TsoaRoute.PropertySchema[] | undefined, parent = ''): any {
+    if (!subSchemas) { return; }
+
+    const subFieldErrors = subSchemas.map(subSchema => {
+      const subFieldErrors: FieldErrors = {};
+      this.ValidateParam(subSchema, value, name, subFieldErrors, parent);
+      return subFieldErrors;
+    });
+
+    if (!subFieldErrors.some(subFieldError => Object.keys(subFieldError).length === 0)) {
+      fieldErrors[parent + name] = {
+        message: `Could not match the union against any of the items. Issues: ${subFieldErrors}`,
+        value,
+      };
+      return;
+    }
+
+    return value;
+  }
+
+  public validateIntersection(name: string, value: any, fieldErrors: FieldErrors, subSchemas: TsoaRoute.PropertySchema[] | undefined, parent = ''): any {
+    if (!subSchemas) { return; }
+
+    const subFieldErrors = subSchemas.map(subSchema => {
+      const subFieldErrors: FieldErrors = {};
+      this.ValidateParam(subSchema, value, name, subFieldErrors, parent);
+      return subFieldErrors;
+    });
+
+    subFieldErrors.forEach(subFieldError => {
+      (Object as any).entries(subFieldError).forEach(([key, value]) => {
+        if (value.message.includes('excess')) {
```

Comment:
Our code shouldn't rely on "magic strings." I know it's unlikely that the error will change as long as I'm one of the owners of the library, but since this line of code depends on the language of the error message thrown from another place, I think we have two options:
1) have the function that returns that error use some kind of statically typed error (like `type IExcessError = Error & {foundExcessErrors:true}`)
2) or probably the easier option, let's just have a unit test that ensures that this bit of code stays in-tact.

Reply:
> Our code shouldn't rely on "magic strings." I know it's unlikely that the error will change as long as I'm one of the owners of the library, but since this line of code depends on the language of the error message thrown from another place

See above.
> E: Especially the way I ignore them right now is not acceptable. I will refactor the additional prop handling, but it somewhat depends on #411.

I noticed I can call validateModel and set additionalProps to false after that gets merged, which I am intending to use so I don't have to do any filtering at all.

Reply:
Sounds good. Since it doesn't look like #411 is going to get much more review anyway, I have merged that so you can continue unencumbered.